lib/AST/Decl.cpp

```diff
@@ -2092,9 +2092,39 @@ bool NominalTypeDecl::derivesProtocolConformance(ProtocolDecl *protocol) const {
     case KnownProtocolKind::BridgedNSError:
       return isObjC() && enumDecl->hasOnlyCasesWithoutAssociatedValues();
 
+    // Enums without associated values and enums with a raw type of String
+    // or Int can explicitly derive CodingKey conformance.
+    case KnownProtocolKind::CodingKey: {
+      Type rawType = enumDecl->getRawType();
+      if (rawType) {
+        auto parentDC = enumDecl->getDeclContext();
+        ASTContext &C = parentDC->getASTContext();
+
+        auto nominal = rawType->getAnyNominal();
+        return nominal == C.getStringDecl() || nominal == C.getIntDecl();
+      } else {
+        // Empty enums are allowed to conform as well.
+        return enumDecl->getAllElements().empty() ||
+               enumDecl->hasOnlyCasesWithoutAssociatedValues();
+      }
+    }
+
     default:
       return false;
     }
+  } else if (isa<StructDecl>(this) || isa<ClassDecl>(this)) {
```

Comment:
When you fix this, can you also add a * to the auto above on knownProtocol. (Or even change it to a reference). I know that this is something that is not in the code that you have touched yourself.